pkg/controller/v1alpha2/applicationconfiguration/apply.go

```diff
@@ -63,19 +63,26 @@ func (a *workloads) Apply(ctx context.Context, status []v1alpha2.WorkloadStatus,
 	// they are all in the same namespace
 	var namespace = w[0].Workload.GetNamespace()
 	for _, wl := range w {
-		if err := a.client.Apply(ctx, wl.Workload, ao...); err != nil {
-			return errors.Wrapf(err, errFmtApplyWorkload, wl.Workload.GetName())
+		if !wl.Unready {
+			err := a.client.Apply(ctx, wl.Workload, ao...)
```

Comment:
I think we can't apply workload before its trait still unready. They should work as a whole.

Reply:
Why do we assume such logic in dependency? What's the reasoning?